test/recipes/30-test_evp_enc.t

```diff
@@ -0,0 +1,80 @@
+#! /usr/bin/env perl
+# Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+#
+# ======================================================================
+# Copyright (c) 2017 Oracle and/or its affiliates.  All rights reserved.
+
+
+use strict;
+use warnings;
+
+use File::Spec::Functions qw/catfile/;
+use File::Copy;
+use File::Compare qw/compare_text/;
+use File::Basename;
+use OpenSSL::Test qw/:DEFAULT srctop_file/;
+
+setup("test_evp_enc");
+
+my $testsrc = srctop_file("test", "recipes", basename($0));
+
+my $cipherlistfile = catfile(".", "ciphermodes");
+my $cipherlist = undef;
+my $plaintext = catfile(".", "testdatafile");
+my $fail = "";
+my $cmd = "openssl";
+
+# The result from enc -ciphers is a failure code, so we have to check
+# the output instead.
+unlink $cipherlistfile;
+run(app([$cmd, "enc", "-ciphers"], stderr => $cipherlistfile,
+                                   stdout => undef));
+{
+    local $/ = undef;
+    if (open F, $cipherlistfile) {
+        $cipherlist = <F>;
+        close F;
+    } else {
+        $fail = "file open : $!";
+    }
+}
+unlink $cipherlistfile;
```

Comment:
Is there a reason why you're going through a file to get all the ciphers?  Wouldn't it be easier (and less error prone) to use the same capture as in `test_enc`?  I.e.:

``` perl
my @ciphers =
    grep { $_ !~ /wrap|hmac|poly|ocb|xts|^$|^[^-]|(?i)[cg]cm/ }
    map { s/^\s+//; s/\s+$//; split /\s+/ }
    run(app([$cmd, "list", "-cipher-commands"]), capture => 1);
```